src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -105,76 +126,105 @@ public Future<ConnectionStatus> getConnectionStatus() {
   }
 
   protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
-    var kafkaConfig = spec.kafkaClusterConfig();
-    if (kafkaConfig == null) {
+    return withAdminClient(adminClient -> {
+      // There is a Kafka configuration, so validate the connection by creating an AdminClient
+      // and describing the cluster.
+      try {
+        var clusterId = adminClient
+            .describeCluster()
+            .clusterId()
+            .get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        return Future.succeededFuture(
+            ConnectionStatusKafkaClusterStatusBuilder
+                .builder()
+                .state(ConnectedState.SUCCESS)
+                .build()
+        );
+      } catch (Exception e) {
+        // The connection failed
+        Throwable cause = e;
+        if (e instanceof java.util.concurrent.ExecutionException) {
+          cause = e.getCause();
+        }
+        // so successfully return the failed state
+        return Future.succeededFuture(
+            ConnectionStatusKafkaClusterStatusBuilder
+                .builder()
+                .state(ConnectedState.FAILED)
+                .errors(
+                    new AuthErrors().withSignIn(
+                        "Failed to connect to Kafka cluster: %s".formatted(cause.getMessage())
+                    )
+                ).build()
+        );
+      }
+    }).orElseGet(() -> {
+      // There is no Kafka cluster configuration, so return the NONE state
```

Comment:
Do you think we should require a Kafka cluster config for Direct connections?

Reply:
I'd prefer to leave it as-is, where you have to have _either_ a Kafka cluster or an SR or both. Although we don't expose them yet, there are some valid use cases for connecting to an SR and just viewing or working with schemas w/o having to work with Kafka topics that use those schemas.